### DISCUSSION TITLE
link with blas and lapack in Intel MKL kit when building IpOpt

### DISCUSSION THREAD

**Discussion by cjl88** [coin-or/Ipopt#453] [Q&A] [answered]
I followed instructions in https://coin-or.github.io/Ipopt/INSTALL.html to link IpOpt against Intel MKL blas and lapack:
-L${MKLROOT}/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lm

However, when I do 'ldd libipopt.so', it shows that it it still linking against system blas and lapack in /use/lib64:
	liblapack.so.3 => /usr/lib64/atlas/liblapack.so.3 (0x00007ff1196cc000)
	libblas.so.3 => /usr/lib64/libblas.so.3 (0x00007ff119475000)
	libgfortran.so.3 => /usr/lib64/libgfortran.so.3 (0x00007ff119183000)
	libmkl_intel_ilp64.so.1 => /nfs/software/intel/oneapi/mkl/latest/lib/intel64/libmkl_intel_ilp64.so.1 (0x00007ff1184e7000)
	libmkl_gnu_thread.so.1 => /nfs/software/intel/oneapi/mkl/latest/lib/intel64/libmkl_gnu_thread.so.1 (0x00007ff1167a2000)
	libmkl_core.so.1 => /nfs/software/intel/oneapi/mkl/latest/lib/intel64/libmkl_core.so.1 (0x00007ff10e7dc000)
	libgomp.so.1 => /usr/lib64/libgomp.so.1 (0x00007ff10e5ce000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007ff10e3b1000)

I want to IpOpt to use blas and lapack in Intel MKL kit.

What am I doing wrong?  Thanks.

**Comment by cjl88** [comment, marked as answer]
Mumps used by IpOpt also has to be compiled with same link option to use Intel MKL. 

Perhaps this can be mentioned in instructions in https://coin-or.github.io/Ipopt/INSTALL.html.

**Reply by svigerske** [reply]
Sure, Ipopt will also depend on dependencies of its dependencies.

So, to confirm, Mumps was that was the one that brought in the system lapack and blas dependencies for you?

**Reply by cjl88** [reply]
yes.
